crates/typed-store/src/rocks/mod.rs

```diff
@@ -474,6 +480,63 @@ pub fn open_cf_opts<P: AsRef<Path>>(
     Ok(rocksdb)
 }
 
+/// Opens a database with options, and a number of column families with individual options that are created if they do not exist.
+pub fn open_cf_opts_secondary<P: AsRef<Path>>(
+    primary_path: P,
+    secondary_path: Option<P>,
+    db_options: Option<rocksdb::Options>,
+    opt_cfs: &[(&str, &rocksdb::Options)],
+) -> Result<Arc<rocksdb::DBWithThreadMode<MultiThreaded>>, TypedStoreError> {
+    // Customize database options
+    let mut options = db_options.unwrap_or_default();
+
+    // One common issue when running tests on Mac is that the default ulimit is too low,
+    // leading to I/O errors such as "Too many open files". Raising fdlimit to bypass it.
+    options.set_max_open_files(-1);
+    /* The table cache is locked for updates and this determines the number
+       of shareds, ie 2^10. Increase in case of lock contentions.
+    */
+
+    let mut opt_cfs: std::collections::HashMap<_, _> = opt_cfs.iter().cloned().collect();
+    let cfs = rocksdb::DBWithThreadMode::<MultiThreaded>::list_cf(&options, &primary_path)
+        .ok()
+        .unwrap_or_default();
+
+    let default_rocksdb_options = rocksdb::Options::default();
+    // Add CFs not explicitly listed
+    for cf_key in cfs.iter() {
+        if !opt_cfs.contains_key(&cf_key[..]) {
+            opt_cfs.insert(cf_key, &default_rocksdb_options);
+        }
+    }
+
+    let primary_path = primary_path.as_ref().to_path_buf();
+    let secondary_path = secondary_path
+        .map(|q| q.as_ref().to_path_buf())
+        .unwrap_or_else(|| {
+            let mut s = primary_path.clone();
+            s.pop();
+            s.push("SECONDARY");
+            s.as_path().to_path_buf()
```

Comment:
You could use the [`std::env!` macro](https://doc.rust-lang.org/std/macro.env.html) to allow overriding this as a one-liner.

Reply:
That'll cause override at compile time. Why would we want that?

Reply:
🤦 compile time overrides, not the best idea.
I meant trying to probe the value at runtime. Or a launch argument. Or something other configuration that allows not using the hard coded string.